### PR TITLE
Description for test suites

### PR DESCRIPTION
**Pull request by okurz**
Add API CRUD operations for a description field for test suites, machines, products.

Also deletes 'variables' field which has not been in use for test suites,
machines, products for quite some time now.

Add UI elements for CRUD on test suite description.

Show description as popover on each test row.

Screenshot how the test suite description shows up in the admin table:
openqa_wrapped_test_suite_table_and_description

Screenshot of edit:
openqa_wrapped_test_suite_table_and_description_edit

Screenshot of popover in test overview with content as configured in the test suites database (i.e. over admin view):
openqa_test_suite_description_hover_on_overview2

### CODE REVIEW

**Comment by okurz** [comment]
@aaannz @coolo feedback?

File: templates/test/overview.html.ep [hunk 1]
```diff
@@ -94,7 +94,11 @@
                 % for my $config (@$configs) {
                     % next unless $results->{$config}{$type};
                     <tr>
-                        <td class="name"><%= trim_text($config, 30) %></td>
+                        <!--<td class="name"><%= trim_text($config, $name_limit) %></td>-->
+                        % my $description = 'TODO fill with test suite description. But where to get it from without joining the test suite table onto the job results?';
```
Comment (Martchus) [inline templates/test/overview.html.ep]: If you don't want a join slowing down loading the page, you can find the associated test suite via async AJAX call on popover. Note that the `MACHINE` column might be appended here, eg `btrfs@s390x-zVM-ctc`.

**Comment by okurz** [comment]
> Open question: How to get the actual content from the test suite table description field onto the job results?
>
> 1. Should we join the test suite table into results?
> 2. Explicitly fetching test suites database for the description
> 3. Back to my initial approach of setting just a special variable in the test suite which is then passed onto every job

Solved by prefetching the test suites database querying for all displayed job results in the controller.

Updated screenshot.

This PR now includes https://github.com/os-autoinst/openQA/pull/1040 for the `test#overview` refactoring but if necessary the "support multiple groups" can be skipped for now.

**Comment by okurz** [comment]
@Martchus please review as you worked on parts of it together with me

File: lib/OpenQA/WebAPI/Controller/Admin/Machine.pm [hunk 2]
```diff
@@ -19,9 +19,7 @@ use strict;
 use parent 'OpenQA::WebAPI::Controller::Admin::Table';
 
 sub index {
-    my $self = shift;
-
-    $self->SUPER::admintable('MachineSettings', 'machine');
+    shift->SUPER::admintable('machine');
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Controller/Admin/Machine.pm]: Maybe we can remove this entire method (also in `Product.pm` and `TestSuite.pm`).

File: t/24-worker.t [hunk 3]
```diff
@@ -36,7 +36,7 @@ like(
     exception {
         OpenQA::Worker::Common::api_init(
             {
-                host => 'http://openqa.opensuse.org',
+                host => 'http://any_host',
```
Comment (Martchus) [inline t/24-worker.t]: So this actually solved the problem (when running locally).

Comment (okurz) [inline t/24-worker.t]: yes

File: lib/OpenQA/WebAPI/Controller/Admin/Machine.pm (hunk 2 above)
Comment (okurz) [inline lib/OpenQA/WebAPI/Controller/Admin/Machine.pm]: I will keep this local for later

**Comment by okurz** [comment]
> Also it doesn't look good when there is no description to be shown.

Yes, so I will not show a link when there is no description

**Comment by okurz** [comment]
updated:
* tests fixed (at least locally)
* only show link on test name if there is a description

**Comment by okurz** [comment]
@coolo any chance you are going to merge this PR or https://github.com/os-autoinst/openQA/pull/1040 soon so that I know on what I can base on?

**Comment by okurz** [comment]
fixed word wrapping in title of popover.

Also added more test and tested against production database dump.

File: lib/OpenQA/WebAPI/Controller/Test.pm [hunk 4]
```diff
@@ -475,41 +446,81 @@ sub overview {
...
             && $preferred_machines->{$job->ARCH} ne $job->MACHINE)
         {
             $test .= "@" . $job->MACHINE;
         }
-        push(@configs, $test) unless (grep { $test eq $_ } @configs);
-        $archs{$flavor} = [] unless $archs{$flavor};
+        # poor mans set
+        $archs{$flavor} //= [];
+        # poor mans set
+        $archs{$flavor} //= [];
```
Comment (aaannz) [inline lib/OpenQA/WebAPI/Controller/Test.pm]: This is copy paste typo, right?

Comment (okurz) [inline lib/OpenQA/WebAPI/Controller/Test.pm]: looks more like rebase mistake. fixed.

**Pull request merged**